generators/client/templates/react/src/main/webapp/app/modules/login/logout.tsx.ejs

```diff
@@ -27,29 +27,26 @@ export interface ILogoutProps extends StateProps, DispatchProps {
   logoutUrl: string;
 }
 
-export class Logout extends React.Component<ILogoutProps> {
+export const Logout = (props: ILogoutProps) => {
+  useEffect(() => {
+    props.logout();
+  });
 
-  componentDidMount() {
-    this.props.logout();
+  const logoutUrl = props.logoutUrl;
+  if (logoutUrl) {
+    // if Keycloak, logoutUrl has protocol/openid-connect in it
+    window.location.href =
+      logoutUrl.includes('/protocol')
+        ? logoutUrl + '?redirect_uri=' + window.location.origin
+        : logoutUrl + '?id_token_hint=' + props.idToken + '&post_logout_redirect_uri=' + window.location.origin;
   }
```

Comment:
@qmonmert: wouldn't it be better to add these lines of code in the useEffect function, as the actions on a component lifecycle event are specified in this function ?